src/AmqpChannel.php

```diff
@@ -359,7 +378,13 @@ private function declareQueue()
     private function reconnect()
     {
         $this->retry--;
-        $this->disconnect();
+
+        try {
+            $this->disconnect();
+        }  catch (AMQPProtocolChannelException $e) {
+            // just continue with reconnect
+        }
+
         $this->connect();
```

Comment:
This isn't covered in any tests is it? If you can think of a way to test an exception throw scenario that'd be grand, if not... we'll live ;)

Reply:
currently we cannot mock AmqpChannel, as the are problems with visibility, and the objects could be created to be more friendly to mocking.

For us to test this and mock we should rebuild the AmqpChannel...